Camera2SlowMotion/app/src/main/java/com/example/android/camera2/slowmo/LagCalculator.java

```diff
@@ -0,0 +1,51 @@
+package com.example.android.camera2.slowmo;
+
+import android.content.ContentValues;
+import android.util.Log;
+import java.util.ArrayList;
+
+/**
+ * Calculates lag of each key-press on the server using the key-press timestamps from server and
+ * character seen timestamp from recorded video.
+ */
+public class LagCalculator {
+  private VideoProcessor videoData;
+  private ServerHandler serverData;
+  public ArrayList<Long> lagResult = new ArrayList<Long>();
+
+  public LagCalculator(VideoProcessor videoData, ServerHandler serverData) {
+    this.videoData = videoData;
+    this.serverData = serverData;
+  }
+
+  public ArrayList<Long> calculateLag() {
+    if (serverData.getServerTimestamps().size() < 2 || videoData.getResultsOCR().size() < 2) {
+      Log.d(ContentValues.TAG, "No results to show. Insufficient server-host data.");
+      return null;
+    }
+    Log.d(ContentValues.TAG, "Video start timestamp:" + videoData.getRecordStartTime());
+    // TODO: Server sequence is hardcoded. Add functionality to receive serverSequence.
+    String serverSequence = "m";
+    long serverHostSyncOffset =
+        serverData.getServerStartTimestamp() - serverData.getHostSyncTimestamp();
+    Log.d(ContentValues.TAG, "Server-Host Sync offset:" + serverHostSyncOffset);
+
+    ArrayList<Long> serverTimestamp = serverData.getServerTimestamps();
+    ArrayList<Long> videoFrameTimestamp = videoData.getVideoFrameTimestamp();
+    ArrayList<String> resultsOCR = videoData.getResultsOCR();
```

Comment:
How is multiple text detected by OCR in a single frame handled?  We need to be resilient to this in the final app.  Doesn't need to be in this PR, but please start thinking about it.